aws/resource_aws_elasticsearch_domain.go

```diff
@@ -384,6 +411,21 @@ func resourceAwsElasticSearchDomainCreate(d *schema.ResourceData, meta interface
 		}
 	}
 
+	if v, ok := d.GetOk("cognito_options"); ok {
+
+		options := v.([]interface{})
+		if len(options) > 1 {
```

Comment:
Nitpick: This error checking is already performed by the attribute schema so it can be removed 👍 
* `MaxItems: 1` handles `len(options) > 1`
* `Required: true` on nested arguments handles `options[0] == nil`

I'd recommend going with the below to simplify this:

```go
if v, ok := d.GetOk("cognito_options"); ok && len(v.([]interface{})) > 0 {
  m := v.([]interface{})[0].(map[string]interface{})
  input.CognitoOptions = expandESCognitoOptions(m)
}
```

Reply:
Oh cool! That's much cleaner!